uber/menu.py

```diff
@@ -0,0 +1,51 @@
+from uber.common import *
+
+
+class MenuItem:
+    access = None   # list of permission levels allowed to display this menu
+    href = None     # link to render
+    submenu = None  # submenu to show
+    name = None     # name of Menu item to show
+
+    def __init__(self, href=None, access=None, submenu=None, name=None):
+        if submenu:
+            self.submenu = listify(submenu)
+        else:
+            self.href = href
+
+        self.name = name
+        self.access = access
+
+    def __getitem__(self, key):
+        for sm in self.submenu:
+            if sm.name == key:
+                return sm
+
+    def append_menu_item(self, m):
+        # if we aren't a submenu, convert us to one now
+        if not self.submenu and self.href:
+            self.submenu = [MenuItem(name=self.name, href=self.href)]
+            self.href = None
+
+        self.submenu.append(m)
+
+
+c.MENU = MenuItem(name='Root', submenu=[
```

Comment:
Additionally, I just realized that 'top-level' menu items count as one big submenu, which is... a bit unintuitive. Do you think it's worth it to create a `Menu` class, then make c.MENU a Menu class object that gets a collection of MenuItems? Or is that unnecessary?